contracts/LockedFund.sol

```diff
@@ -0,0 +1,510 @@
+		_withdrawWaitedUnlockedBalance(msg.sender, _receiverAddress);
+		_createVestingAndStake(msg.sender);
+	}
+
+	/**
+	 * @notice Withdraws unlocked tokens and Stakes Locked tokens for a user who already have a vesting created.
+	 * @param _userAddress The address of user tokens will be withdrawn.
+	 */
+	function withdrawAndStakeTokensFrom(address _userAddress) external {
+		_withdrawWaitedUnlockedBalance(_userAddress, _userAddress);
+		_createVestingAndStake(_userAddress);
+	}
+
+	/* Internal Functions */
+
+	/**
+	 * @notice Internal function to add a new admin.
+	 * @param _newAdmin The address of the new admin.
+	 */
+	function _addAdmin(address _newAdmin) internal {
+		require(_newAdmin != address(0), "LockedFund: Invalid Address.");
+		require(!isAdmin[_newAdmin], "LockedFund: Address is already admin.");
+		isAdmin[_newAdmin] = true;
+
+		emit AdminAdded(msg.sender, _newAdmin);
+	}
+
+	/**
+	 * @notice Internal function to remove an admin.
+	 * @param _adminToRemove The address of the admin which should be removed.
+	 * @dev Only callable by an Admin.
+	 */
+	function _removeAdmin(address _adminToRemove) internal {
+		require(isAdmin[_adminToRemove], "LockedFund: Address is not an admin.");
+		isAdmin[_adminToRemove] = false;
+
+		emit AdminRemoved(msg.sender, _adminToRemove);
+	}
+
+	/**
+	 * @notice Internal function to update the Vesting Registry, Duration and Cliff.
+	 * @param _vestingRegistry The Vesting Registry Address.
+	 */
+	function _changeVestingRegistry(address _vestingRegistry) internal {
+		require(_vestingRegistry != address(0), "LockedFund: Vesting registry address is invalid.");
+
+		vestingRegistry = IVestingRegistry(_vestingRegistry);
+
+		emit VestingRegistryUpdated(msg.sender, _vestingRegistry);
+	}
+
+	/**
+	 * @notice Internal function used to update the waitedTS.
+	 * @param _waitedTS The timestamp after which withdrawal is allowed.
+	 */
+	function _changeWaitedTS(uint256 _waitedTS) internal {
+		require(_waitedTS != 0, "LockedFund: Waited TS cannot be zero.");
+
+		waitedTS = _waitedTS;
+
+		emit WaitedTSUpdated(msg.sender, _waitedTS);
+	}
+
+	/**
+	 * @notice Internal function to add Token to the user balance (Vested and Waited Unlocked Balance based on `_basisPoint`).
+	 * @param _userAddress The user whose locked balance has to be updated with `_amount`.
+	 * @param _amount The amount of Token to be added to the locked and/or unlocked balance.
+	 * @param _cliff The cliff for vesting.
+	 * @param _duration The duration for vesting.
+	 * @param _basisPoint The % (in Basis Point)which determines how much will be unlocked immediately.
+	 */
+	function _depositVested(
+		address _userAddress,
+		uint256 _amount,
+		uint256 _cliff,
+		uint256 _duration,
+		uint256 _basisPoint
+	) internal {
+		/// If duration is also zero, then it is similar to Unlocked Token.
+		require(_duration != 0, "LockedFund: Duration cannot be zero.");
+		require(_duration < MAX_DURATION, "LockedFund: Duration is too long.");
```

Comment:
it's called `MAX_DURATION`, but it cannot ever be achieved... and magic number `37` is bit too magical (subjectively).
I'd recommend changing it to `36` and `<=` respectively

Reply:
Done